buildifier/buildifier.go

```diff
@@ -368,6 +373,28 @@ func processFile(filename string, data []byte) {
 	}
 }
 
+func setFormattingMode(inputType, filename string) func() {
+	defaultMode := tables.FormattingMode
+	
+	switch inputType {
+	case "build":
+		tables.FormattingMode = tables.BuildMode
+	case "bzl":
+		tables.FormattingMode = tables.DefaultMode
+	case "auto":
+		base := filepath.Base(filename)
+		if base == "BUILD" || base == "BUILD.bazel" || base == "WORKSPACE" {
+			tables.FormattingMode = tables.BuildMode
+		} else if strings.HasSuffix(filename, ".bzl") {
```

Comment:
@laurentlb should it check for the .bzl extension or should it just use the bzl mode if the filename is not equal to `BUILD` or `WORKSPACE`?

Reply:
you can decide

Reply:
Refactored.